lib/resque/failure/redis.rb

```diff
@@ -48,14 +48,18 @@ def self.all(offset = 0, limit = 1, queue = nil)
       end
 
       def self.each(offset = 0, limit = self.count, queue = :failed, class_name = nil, order = 'desc')
+        if class_name
+          original_limit = limit
+          limit = count
+        end
         all_items = limit == 1 ? [all(offset,limit,queue)] : Array(all(offset, limit, queue))
```

Comment:
off topic, but
* `Array(` doesn't do anything when `all` returns an array
* `all` should just always return an array and there should be another method to get a single item from the queue, so this can just be `all_items = all(offset, limit, queue)`